### PR TITLE
populate files slice on multipart/form submission

### PR DESCRIPTION
**Pull request by llimllib**
- populate files slice on multipart/form-data request
  - it was already present, but just wasn't filled out anywhere
- Is my test sufficient?
- How I'm currently integration testing this is by running this script:

```bash
printf "Hello World\n" > hello.txt
curl -s --request POST \
  --url http://0.0.0.0:8080/anything \
  --header 'content-type: multipart/form-data' \
  --form foo=@hello.txt \
  --form 'bar=Bonjour le monde' | jq .files

```

against a development server and expecting this output:

```json
{
  "foo": [
    "Hello World\n"
  ]
},
```

### CODE REVIEW

**Review by mccutchen — COMMENTED** [review]
Looks good to me so far!

> Is my test sufficient?

Left one very minor suggestion, feel free to ignore it.

It would be nice to add a test that generates an invalid multipart/form submission, just to have a negative test case and try to exercise some of the new error handling, but I'm afraid that might be more hassle than it's worth.  (In particular, my guess is that it might be hard to construct a request that will pass the initial request form parsing but fail once `parseFiles()` tries to open or read files from the incoming request?)

File: httpbin/handlers_test.go [hunk 1]
```diff
@@ -816,6 +817,51 @@ func testRequestWithBodyMultiPartBody(t *testing.T, verb, path string) {
...
+	if bodies, ok := resp.Files["fieldname"]; ok {
+		if len(bodies) != 1 {
+			t.Fatalf("expected one file, got %v", bodies)
+		}
+		if bodies[0] != "hello world" {
+			t.Fatalf("expected the body to be \"hello world\" but got %s", bodies[0])
+		}
+	} else {
+		t.Fatalf("expected to have a file called \"fieldname\" in %v", resp.Files)
+	}
```
[the remark this answers is not in the document]
Comment (llimllib) [reply httpbin/handlers_test.go]: Works for me, updated the tests.

I usually reach for [go-cmp](https://github.com/google/go-cmp) when I start doing map comparisons so you get nice diffs on errors, though that brings its own complications with it.

I tried briefly to come up with a way to make `parseFiles` throw either of the two error cases, and I didn't come up with anything clever. [Here's the go source tests](https://github.com/golang/go/blob/413e6c0499cfc29a0306ebc27160eca32882089c/src/mime/multipart/formdata_test.go#L108) for form parsing; sometimes I read their tests and get good ideas but this time my eyes just kind of glazed over tbh

[the remark this answers is not in the document]
Comment (mccutchen) [reply httpbin/handlers_test.go]: Yeah, that's understandable, I'm comfortable merging this as-is.  Thanks!
